Tests/JSONAPITests/ResourceTests.swift

```diff
@@ -112,6 +112,17 @@ final class ResourceTests: XCTestCase {
 		XCTAssertEqual(article, Fixtures.article)
 	}
 
+	func testDecodeNullSingle() throws {
+		// given
+		let json = try XCTUnwrap(#"{ "data": null }"#.data(using: .utf8))
+
+		// when
+		let article = try JSONAPIDecoder().decode(Article?.self, from: json)
+
+		// then
+		XCTAssertNil(article)
+	}
```

Comment:
It could be nice to add a test case for the optional overload but with an actual resource instead of `null`.